FIX: auth in URL causes bad requests

SamSaffron opened the pull request:
Strip user and password from root, otherwise it confuses HTTP requests
which need to be explicit about basic auth

SamSaffron committed:
FIX: auth in URL causes bad requests

Strip user and password from root, otherwise it confuses HTTP requests
which need to be explicit about basic auth

SamSaffron commented:
There is a test here, but somehow it is not failing... maybe the stubbing is off a bit? 

https://github.com/discourse/discourse_theme/blob/f9f402d86732be14ada0642b6fee4bf2527bd9cd/test/test_cli.rb#L68-L87

It is certainly failing in production.

SamSaffron commented:
Kind of giving up my journey down webmock land. 

It internally adapts the HTTP::URI into an Addressable::URI and strips out user and password... so yay... can not find an easy way even after upgrading webmock to test this.

Our surface area is pretty small so I guess we could build our own abstraction here for testing, but I am not sure it is worth the time.

nattsw commented:
You may want to bump `lib/discourse_theme/version.rb` as well to get the gem published when merged.

SamSaffron committed:
update webmock and add m

usage `bundle exec rake m test/test_cli:83`

Allows you to run single tests easily

(used it while debugging tests, but it go nowhere, since webmock adapts
objects and throws away some info)

SamSaffron committed:
bump version
